flocker/node/agents/blockdevice.py

```diff
@@ -1470,6 +1722,13 @@ def discover_state(self, node_state):
                         maximum_size=volume.size,
                         blockdevice_id=volume.blockdevice_id,
                     )
+                else:
```

Comment:
This will record datasets with `attached_to == None` as ATTACHED_ELSEWHERE, which is wrong.